src/main/scala/nl/vroste/zio/kinesis/client/DynamicConsumerFake.scala

```diff
@@ -0,0 +1,135 @@
+package nl.vroste.zio.kinesis.client
+
+import java.nio.ByteBuffer
+
+import nl.vroste.zio.kinesis.client.DynamicConsumer.Checkpointer
+import nl.vroste.zio.kinesis.client.serde.{ Deserializer, Serializer }
+import software.amazon.awssdk.services.kinesis.model.EncryptionType
+import software.amazon.kinesis.common.InitialPositionInStreamExtended
+import zio._
+import zio.blocking.Blocking
+import zio.clock.Clock
+import zio.stream.ZStream
+
+private[client] class DynamicConsumerFake(
+  shards: ZStream[Any, Throwable, (String, ZStream[Any, Throwable, ByteBuffer])],
+  refCheckpointedList: Ref[Seq[Any]],
+  clock: Clock.Service
+) extends DynamicConsumer.Service {
+  override def shardedStream[R, T](
+    streamName: String,
+    applicationName: String,
+    deserializer: Deserializer[R, T],
+    requestShutdown: UIO[Unit],
+    initialPosition: InitialPositionInStreamExtended,
+    isEnhancedFanOut: Boolean,
+    leaseTableName: Option[String],
+    workerIdentifier: String,
+    maxShardBufferSize: Int
+  ): ZStream[
+    Blocking with R,
+    Throwable,
+    (String, ZStream[Blocking, Throwable, Record[T]], DynamicConsumer.Checkpointer)
+  ] = {
+    def record(shardName: String, i: Long, recData: T): UIO[Record[T]] =
+      (for {
+        dateTime <- clock.currentDateTime
+      } yield new Record[T](
+        sequenceNumber = s"$i",
+        approximateArrivalTimestamp = dateTime.toInstant,
+        data = recData,
+        partitionKey = s"${shardName}_$i",
+        encryptionType = EncryptionType.NONE,
+        subSequenceNumber = Some(i),
+        explicitHashKey = None,
+        aggregated = false,
+        shardId = shardName
+      )).orDie
+
+    shards.flatMap {
+      case (shardName, stream) =>
+        ZStream.fromEffect {
+          ZIO.environment[R with Blocking].flatMap { env =>
+            CheckpointerFake.make(refCheckpointedList).map { checkpointer =>
+              (
+                shardName,
+                stream.zipWithIndex.mapM {
```

Comment:
There's an annoying issue in ZIO streams that makes `mapM` really slow and the recommendation is to use `mapChunksM`. I don't think it's necessary for this fake implementation, but just fyi.

https://github.com/zio/zio/pull/3666

Reply:
:+1: